src/api/app/presenters/repositories/state_presenter.rb

```diff
@@ -0,0 +1,11 @@
+module Repositories
+  class StatePresenter < SimpleDelegator
+    attr_reader :published_repository_exist, :download_area_url
+
+    def initialize(project, repository, configuration)
+      download_url = configuration['download_url']
```

Comment:
This can be `nil`. See https://obs-reviewlab.opensuse.org/danidoni-render_download_repository_link_only_if_published_artifacts_exist/project/repository_state/home:Admin/openSUSE_Leap_15.2

Reply:
Ok, I'll make `@published_repository_exist` to be false when there is no `download_url` configured.

Reply:
should everything happen in the `initialize` method?

Reply:
should everything happen in the initialize method?

Reply:
Why not? You would delay the evaluation in the calling methods if the computation can change from call to call, but I'm doing it because the data passed as configuration won't change, ever. So there is no reason to lazy load it. 

You could say it's a matter of taste. And you would be right. I have no problem in changing it if there is a technical reason, as you know :)